Adds versioning changes to the `applies_to` page (#1325)
elastic/docs-builder: docs/syntax/applies.md

```

Taking a mandatory [life-cycle](#life-cycle) with an optional version.

### Version

Can be in either `major.minor` or `major.minor.patch` format

Versioned products require a `version` tag to be used with the `lifecycle` tag. See [Syntax](#syntax):

```
applies_to:
  stack: preview 9.1, ga 9.4
  deployment:
    ece: deprecated 9.2, removed 9.8
```
Unversioned products use `lifecycle` tags without a version:

```
applies_to:
  serverless:
    elasticsearch: beta
    observability: removed
```

### Combined states
You can specify multiple lifecycle states for the same product, separated by commas. For example:

```
applies_to:
  stack: preview 9.1, ga 9.4
```
This shows that the feature was introduced in version 9.1 as a preview and became generally available in 9.4.

### Life cycle

`applies_to` accepts the following lifecycle states:

  * `preview`
  * `beta`
  * `deprecated`
  * `removed`
  * `unavailable`
  * `ga`

Both versioned and unversioned products use the same lifecycle tags, but only versioned products can be marked `ga`. Unversioned products are considered `ga` by default and don’t need specification.

## When and where to use `applies_to`

> The `applies_to` tag can be added at different levels in the documentation: [page-level](#page-annotations), [section-level](#section-annotations), and [inline](#inline-applies-to). Each level uses slightly different syntax and serves a specific purpose:
> * [Page-level](#page-annotations) tagging is **mandatory** and must be included in the frontmatter. It defines the overall applicability of the page across products, deployments, and environments. For a complete list of supported keys and values, see the [frontmatter syntax guide](./frontmatter.md).
> 
> When the context differs from what was specified at the page level in a specific section or part of the page, it is appropriate to re-establish it.
> 
> * [Section-level](#section-annotations) annotation lets you show or hide specific sections of content depending on the target context. This is helpful when only a part of a page varies between products or versions.
> * [Inline tagging](#inline-applies-to) allows fine-grained annotations within paragraphs or definition lists. It’s useful for highlighting the applicability of specific phrases, sentences, or properties without disrupting the surrounding content.
> 
> ### When to use `applies_to`
> 
> Every page must include a [page-level `applies_to`](#page-annotations) tag to clearly define its scope and availability.
> 
> Use version tagging when:
> * A feature is introduced (e.g., preview, beta, or ga)
> * A feature is deprecated (e.g., deprecated)
> * A feature is removed (e.g., removed)
> 
> You don’t need version tagging for:
> * Typos, formatting, or style changes
> * Long-standing features being documented for the first time
> * Content updates that don’t reflect a feature lifecycle change

Reviewer: ```suggestion
✅ Use `applies_to` tags when features change state (`introduced`, `deprecated`, `removed`) or when availability differs across deployments and environments.

❌ Don't tag content-only changes like typos, formatting, or documentation updates that don't reflect feature lifecycle changes.

The `applies_to` metadata can be added at different levels in the documentation: 

* [Page-level](#page-annotations) is **mandatory** and must be included in the frontmatter. It defines the overall applicability of the page across products, deployments, and environments.
* [Section-level](#section-annotations) annotations allow you to specify different applicability for individual sections when only part of a page varies between products or versions.
* [Inline](#inline-applies-to) annotations allow fine-grained annotations within paragraphs or definition lists. This is useful for highlighting the applicability of specific phrases, sentences, or properties without disrupting the surrounding content.
```

This improves section flow by leading with clear usage guidelines before diving into implementation mechanics, eliminating redundant nested headings

### Page annotations

> All documentation pages **must** include an `applies_to` tag in the YAML frontmatter. Use yaml frontmatter to indicate each deployment targets availability and lifecycle status.

Reviewer: ```suggestion
All documentation pages **must** include an `applies_to` tag in the YAML frontmatter. Use yaml frontmatter to indicate each deployment targets availability and lifecycle status. For a complete list of supported keys and values, see the [frontmatter syntax guide](./frontmatter.md).
```

tied to https://github.com/elastic/docs-builder/pull/1325/files#r2120864141

Taking a mandatory [life-cycle](#life-cycle) with an optional version.

#### Life cycle

Both versioned and unversioned products use the same lifecycle tags, but only versioned products can be marked `ga`. Unversioned products are considered `ga` by default and don’t need specification.

  * `preview`
  * `beta`
  * `deprecated`
  * `removed`
  * `unavailable`
  * `ga`

#### Version

Can be in either `major.minor` or `major.minor.patch` format

Versioned products require a `version` tag to be used with the `lifecycle` tag. See [Syntax](#syntax):

```
applies_to:
  stack: preview 9.1, ga 9.4
  deployment:
    ece: deprecated 9.2, removed 9.8
```
Unversioned products use `lifecycle` tags without a version:

```
applies_to:
  serverless:
    elasticsearch: beta
    observability: removed
```

#### Examples

```
preview 9.5, ga 9.7
deprecated 9.9.0
unavailable
```

> ## When to use `applies_to`

Reviewer: At second glance, I think this section is kind of thin and could potentially be removed entirely and the guidance for "Tag feature lifecycle changes" and 'Don't tag content-only changes' could be folded into the new section about different levels
